flow-server/src/main/java/com/vaadin/flow/server/VaadinServletService.java

```
@@ -174,6 +178,19 @@ protected RouteRegistry getRouteRegistry() {
         return RouteRegistry.getInstance(getServlet().getServletContext());
     }
 
+    @Override
+    protected PWARegistry getPwaRegistry() {
+        if (pwaRegistry == null) {
+            try {
+                this.pwaRegistry = PWARegistry
+                        .initRegistry(getServlet().getServletContext());
+            } catch (IOException e) {
+                return null;
```

Instead of returning null throw a new `NamedException extends RuntimeException` with information on what failed and why as now things will fail with a NPE at some point.